jl4-lsp/src/LSP/L4/Viz/Ladder.hs

```diff
@@ -221,19 +289,18 @@ scanOr e = [e]
 defaultUBoolVarValue :: V.UBoolValue
 defaultUBoolVarValue = V.UnknownV
 
-leafFromVizName :: V.Name -> Viz IRExpr
-leafFromVizName vname = do
-  uid <- getFresh
-  pure $ V.UBoolVar uid vname defaultUBoolVarValue
+leafFromVizName :: V.ID -> V.Name -> Viz IRExpr
+leafFromVizName vid vname = do
+  pure $ V.UBoolVar vid vname defaultUBoolVarValue
```

Comment:
This function should probably not be effectful, if it doesn't even "conceptually" require to be

Reply:
Nice catch!